pkg/controller/devopsproject/devopsproject_controller.go

```diff
@@ -312,6 +326,38 @@ func (c *Controller) syncHandler(key string) error {
 	return nil
 }
 
+func (c *Controller) bindWorkspace(project *devopsv1alpha3.DevOpsProject) (*devopsv1alpha3.DevOpsProject, error) {
+
+	workspaceName := project.Labels[constants.WorkspaceLabelKey]
+
+	if workspaceName == "" {
+		return project, nil
+	}
+
+	workspace, err := c.workspaceLister.Get(workspaceName)
+
+	if err != nil {
+		// skip if workspace not found
+		if errors.IsNotFound(err) {
+			return project, nil
+		}
+		klog.Error(err)
+		return nil, err
+	}
+
+	if !metav1.IsControlledBy(project, workspace) {
+		project.OwnerReferences = nil
```

Comment:
Why we `OwnerReferences` to nil first, what if there is already a ownerReference?

Reply:
Make sure devops project is controlled by the specified workspace, resolve `AlreadyOwnedError` if already bind another workspace.